src/gwt/src/org/rstudio/studio/client/common/shell/ShellWidget.java

```diff
@@ -132,27 +132,50 @@ public void onKeyDown(KeyDownEvent event)
             if (input_.isPopupVisible())
                return;
             
-            // If the user hits Page-Up from inside the console input, we need
-            // to simulate pageup because focus is not contained in the scroll
-            // panel (it's in the hidden textarea that Ace uses under the
-            // covers).
+            // If the user hits PageUp or PageDown from inside the console
+            // input, we need to simulate its action because focus is not contained
+            // in the scroll panel (it's in the hidden textarea that Ace uses
+            // under the covers).
 
             int keyCode = event.getNativeKeyCode();
             switch (keyCode)
             {
                case KeyCodes.KEY_PAGEUP:
+               {
                   event.stopPropagation();
                   event.preventDefault();
 
                   // Can't scroll any further up. Return before we change focus.
                   if (scrollPanel_.getVerticalScrollPosition() == 0)
                      return;
 
+                  int newScrollTop =
+                        scrollPanel_.getVerticalScrollPosition() -
+                        scrollPanel_.getOffsetHeight() +
+                        40;
+                  
                   scrollPanel_.focus();
-                  int newScrollTop = scrollPanel_.getVerticalScrollPosition() -
-                                     scrollPanel_.getOffsetHeight() + 40;
                   scrollPanel_.setVerticalScrollPosition(Math.max(0, newScrollTop));
                   break;
+               }
+                  
+               case KeyCodes.KEY_PAGEDOWN:
+               {
+                  event.stopPropagation();
+                  event.preventDefault();
+                  
+                  if (scrollPanel_.isScrolledToBottom())
+                     return;
+                  
+                  int newScrollTop =
+                        scrollPanel_.getVerticalScrollPosition() +
+                        scrollPanel_.getOffsetHeight() -
+                        40;
+                  
+                  scrollPanel_.focus();
```

Comment:
Does this take away user's ability to continuing typing into the console? Anytime I see an explicit `focus()` call there's an automatic keyboard accessibility concern.

Reply:
FWIW, I haven't dug into accessibility of the console pane yet (it's a minefield of potential problems).

Reply:
This appears to work transparently -- when we get a key event while the console history has focus, we re-route focus back to the console input (so that it's "as though" the console input had focus all along)

There's probably further tweaks required to make this properly accessible, but I think this should suffice for now (and at least for v1.2-patch).